lib/trie/proof_test.go

```diff
@@ -17,237 +17,39 @@
-			val, _ := proof.Get(key)
-			proof.Del(key)
-			it.Release()
-
-			newhash, err := common.Keccak256(val)
-			require.NoError(t, err)
-			proof.Put(newhash.ToBytes(), val)
-
-			v, err := VerifyProof(rootHash, kv.k, proof)
-			require.NoError(t, err)
-			require.False(t, v)
-		}(entry)
-	}
-
-	wg.Wait()
-}
-
-func TestGenerateProofEmptyNibbles(t *testing.T) {
-	k := []byte{}
-	trie := NewEmptyTrie()
-	_, err := trie.GenerateProof(k, nil)
-	require.Error(t, err, ErrEmptyNibbles)
-}
-
-func TestGenerateProofNilRoot(t *testing.T) {
-	k := []byte{0xff, 0xff}
-	trie := NewEmptyTrie()
-	_, err := trie.GenerateProof(k, nil)
-
-	require.Error(t, err, "no more paths to follow")
-}
-
-func TestGenerateProofMissingKey(t *testing.T) {
-	trie := NewEmptyTrie()
-
-	parentKey, parentVal := randBytes(32), randBytes(20)
-	chieldKey, chieldValue := modifyLastBytes(parentKey), modifyLastBytes(parentVal)
-	gransonKey, gransonValue := modifyLastBytes(chieldKey), modifyLastBytes(chieldValue)
-
-	trie.Put(parentKey, parentVal)
-	trie.Put(chieldKey, chieldValue)
-	trie.Put(gransonKey, gransonValue)
-
-	proof, clear := inMemoryChainDB(t)
-	defer clear()
-
-	searchfor := make([]byte, len(gransonKey))
-	copy(searchfor[:], gransonKey[:])
-
-	// keep the path til the key but modify the last element
-	searchfor[len(searchfor)-1] = searchfor[len(searchfor)-1] + byte(0xff)
-
-	_, err := trie.GenerateProof(searchfor, proof)
-	require.Error(t, err, "leaf node doest not match the key")
-}
-
-func TestGenerateProofNoMorePathToFollow(t *testing.T) {
-	trie := NewEmptyTrie()
-
-	parentKey, parentVal := randBytes(32), randBytes(20)
-	chieldKey, chieldValue := modifyLastBytes(parentKey), modifyLastBytes(parentVal)
-	gransonKey, gransonValue := modifyLastBytes(chieldKey), modifyLastBytes(chieldValue)
-
-	trie.Put(parentKey, parentVal)
-	trie.Put(chieldKey, chieldValue)
-	trie.Put(gransonKey, gransonValue)
-
-	proof, clear := inMemoryChainDB(t)
-	defer clear()
-
-	searchfor := make([]byte, len(parentKey))
-	copy(searchfor[:], parentKey[:])
-
-	// the keys are equals until the byte number 20 so we modify the byte number 20 to another
-	// value and the branch node will no be able to found the right slot
-	searchfor[20] = searchfor[20] + byte(0xff)
-
-	_, err := trie.GenerateProof(searchfor, proof)
-	require.Error(t, err, "no more paths to follow")
-}
-
-type kv struct {
-	k []byte
-	v []byte
-}
-
-func randomTrie(t *testing.T, n int) (*Trie, map[string]*kv) {
-	t.Helper()
-
-	trie := NewEmptyTrie()
-	vals := make(map[string]*kv)
-
-	for i := 0; i < n; i++ {
-		v := &kv{randBytes(32), randBytes(20)}
-		trie.Put(v.k, v.v)
-		vals[string(v.k)] = v
-	}
-
-	return trie, vals
-}
-
-func randBytes(n int) []byte {
-	r := make([]byte, n)
-	crand.Read(r)
-	return r
-}
-
-func modifyLastBytes(b []byte) []byte {
-	newB := make([]byte, len(b))
-	copy(newB[:], b)
-
-	rb := randBytes(12)
-	copy(newB[20:], rb)
-
-	return newB
+	// TODO: use the verify_proof function to assert the tests
```

Comment:
Is there plans to do this in a different PR?

Reply:
yes! the issue to create the `verify_proof` function is already done!